### PR TITLE
update to nixpkgs 23.05

### PR DESCRIPTION
**Pull request by cdmistman**
Why
===

it's nice having the latest and greatest, and with nix modules we can be sure little will break.

What changed
============

- updated `inputs.nixpkgs` to point to `"github:nixos/nixpkgs/nixos-23.05"` instead of a specific `?rev=`. This will allow us to easily update nixpkgs in the future
- handled deprecations in nodejs 14, 16, and 19
- add nodejs 20
- use `pkgs.phpactor` instead of our own, remove the latter
- add appropriate entries in the upgrade mapping

Test plan
=========

all of the released nix modules work

Rollout
=======

***NOTE*: I'm not sure why this started occurring with 23.05 instead of 22.11, but it seems that the `"license": "UNLICENSED"` in `replbox` means that we now need `NIXPKGS_ALLOW_UNFREE=1` and the `--impure` flag to build the devShell and the bundle**

- [ ] This is fully backward and forward compatible

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,22 +1,21 @@
 {
   description = "Nix expressions for defining Replit development environments";
-  inputs.nixpkgs.url = "github:nixos/nixpkgs?rev=52e3e80afff4b16ccb7c52e9f0f5220552f03d04";
-  inputs.nixpkgs-unstable.url = "github:nixos/nixpkgs?rev=87f9156865ab09e3bde39aadb4131ae364ae704e";
-  inputs.prybar.url = "github:replit/prybar?rev=65f486534054665f1b333689417c39acd370d3a5";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-23.05";
+  inputs.nixpkgs-unstable.url = "github:nixos/nixpkgs/nixpkgs-unstable";
```
Comment (cdmistman) [inline flake.nix:4]: using `?rev` [no longer needed](https://replit.slack.com/archives/C03KS2B221W/p1686178911723559?thread_ts=1686154091.642739&cid=C03KS2B221W)

File: flake.nix [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
+  inputs.prybar.url = "github:replit/prybar";
 
   outputs = { self, nixpkgs, nixpkgs-unstable, prybar, ... }:
     let
-      mkPkgs = system: import nixpkgs {
+      mkPkgs = nixpkgs-spec: system: import nixpkgs-spec {
         inherit system;
         overlays = [ self.overlays.default prybar.overlays.default ]; # ++ import ;
+        # replbox has an unfree license
+        config.allowUnfree = true;
```
Comment (cdmistman) [inline flake.nix:13]: maybe we change the license? or just import directly from the [repo](https://github.com/replit/replbox) idk why it was pulled in

File: pkgs/phpactor/default.nix
Comment (cdmistman) [inline pkgs/phpactor/default.nix:1]: ryan's patch/update was merged upstream

File: pkgs/upgrade-maps/default.nix [hunk 3]
```diff
@@ -8,15 +8,28 @@ let
     #"nodejs-16:v1-20230522-49470df" = { to = "nodejs-16:v2-20230522-4c01fa0"; auto = true; changelog = "improved your experience"; };
     #"nodejs-14:v3-20230522-36692ed" = { to = "nodejs-18:v3-20230522-36692ed"; changelog = "Node.js 14 is deprecated. Upgrade to 18!"; };
     #"nodejs-16:v3-20230522-36692ed" = { to = "nodejs-18:v3-20230522-36692ed"; changelog = "Node.js 16 is deprecated. Upgrade to 18!"; };
-    "go" = { to = "go-1.19:v1-20230525-c48c43c"; auto = true; };
-    "rust" = { to = "rust-1.69:v1-20230525-c48c43c"; auto = true; };
-    "swift" = { to = "swift-5.6:v1-20230525-c48c43c"; auto = true; };
```
Comment (cdmistman) [inline pkgs/upgrade-maps/default.nix:13]: i like alphabetical ordering for maintainability

File: scripts/lock_modules.py
Comment (cdmistman) [inline scripts/lock_modules.py:1]: changes here needed due to replbox :/

File: modules.json
Comment (cdmistman) [inline modules.json:1]: again, alphabetical ordering. kinda annoying that the script only keeps ordering for identical module names which include the community version

**Comment by masad-frost** [comment]
Passing this on to Ryan. I don't have enough context to review, I'll have to collect it at some point but can't right tnow

**Comment by cdmistman** [comment]
ci failing due to <https://github.com/replit/nixmodules/blob/2f897128bcb48a50b6288d22bcee14486c010e44/scripts/moduleit_test.sh#L3>. I'm gonna have to update the script to use `flake.lock` instead (or some `nix flake` command)

File: flake.nix (hunk 1 above)
Comment (ryantm) [reply flake.nix:4]: I suspect we're still going to need it, but happy to try again!

File: scripts/lock_modules.py [hunk 4]
```diff
@@ -56,7 +56,9 @@ def is_working_directory_clean():
   return len(output) == 0
 
 def get_current_modules():
-  output = subprocess.check_output(['nix', 'eval', '.#modules', '--json'] + nix_flags)
+  nix_env = os.environ.copy()
+  nix_env['NIX_ALLOW_UNFREE'] = '1'
```
[the remark this answers is not in the document]
Comment (cdmistman) [reply scripts/lock_modules.py:60]: unfortunately we do, since the script gets run outside of derivations as well.

**Comment by cdmistman** [comment]
Current dependencies on/for this PR:
* main
  * **PR #41** <a href="https://app.graphite.dev/github/pr/replit/nixmodules/41" target="_blank">Graphite</a>  👈
    * **PR #54** <a href="https://app.graphite.dev/github/pr/replit/nixmodules/54" target="_blank">Graphite</a> 
      * **PR #55** <a href="https://app.graphite.dev/github/pr/replit/nixmodules/55" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/replit/nixmodules/41?utm_source=stack-comment).

**Review by ryantm — APPROVED** [review]
looks good now :rocket:

**Pull request merged**